_posts/2019-05-21-accessing-kafka-part-5.md

```diff
@@ -0,0 +1,218 @@
+```
+
+For example:
+
+```
+bin/kafka-console-producer.sh --broker-list bootstrap.192.168.64.46.nip.io:443 --producer-property security.protocol=SSL --producer-property ssl.truststore.password=password --producer-property ssl.truststore.location=./truststore.jks --topic <your-topic>
+```
+
+For more details, see the [Strimzi documentation](https://strimzi.io/docs/master/full.html#proc-accessing-kafka-using-ingress-deployment-configuration-kafka).
+
+## Customizations
+
+### DNS annotations
+
+Many users are using some additional tools such as [ExternalDNS](https://github.com/kubernetes-incubator/external-dns) to automatically manage DNS records for their load balancers.
+External DNS uses annotations on `Ingress` resources to manage their DNS names.
+It supports many different DNS services such as Amazon AWS Route53, Google Cloud DNS, Azure DNS etc.
+
+Strimzi lets you assign these annotations through the `Kafka` custom resource using a field called `dnsAnnotations`.
+Using the DNS annotations is simple:
+
+```yaml
+# ...
+listeners:
+  external:
+    type: ingress
+    configuration:
+      bootstrap:
+        host: kafka-bootstrap.mydomain.com
+      brokers:
+      - broker: 0
+        host: kafka-broker-0.mydomain.com
+      - broker: 1
+        host: kafka-broker-1.mydomain.com
+      - broker: 2
+        host: kafka-broker-2.mydomain.com
+    overrides:
+      bootstrap:
+        dnsAnnotations:
+          external-dns.alpha.kubernetes.io/hostname: kafka-bootstrap.mydomain.com.
+          external-dns.alpha.kubernetes.io/ttl: "60"
+      brokers:
+      - broker: 0
+        dnsAnnotations:
+          external-dns.alpha.kubernetes.io/hostname: kafka-broker-0.mydomain.com.
+          external-dns.alpha.kubernetes.io/ttl: "60"
+      - broker: 1
+        dnsAnnotations:
+          external-dns.alpha.kubernetes.io/hostname: kafka-broker-1.mydomain.com.
+          external-dns.alpha.kubernetes.io/ttl: "60"
+      - broker: 2
+        dnsAnnotations:
+          external-dns.alpha.kubernetes.io/hostname: kafka-broker-2.mydomain.com.
+          external-dns.alpha.kubernetes.io/ttl: "60"
+# ...
+```
+
+Again, Strimzi lets you configure the annotations directly.
+That gives you more freedom and hopefully makes this feature usable even when you use some other tool then External DNS.
+It also lets you configure other options than just the DNS names, such as the time-to-live of the DNS records etc.
+
+## Pros and cons
+
+Kubernetes Ingress is not always easy to use because you have to install the Ingress controller and you have to configure the hosts.
+It is also available only with TLS encryption because of the TLS passthrough functionality which Strimzi uses.
+But it can offer an interesting option for clusters where node ports are not an option for example for security reasons and using load balancers would be too expensive.
```

Comment:
```suggestion
But it can offer an interesting option for clusters where node ports are not an option, for example for security reasons, and where using load balancers would be too expensive.
```